### ISSUE TITLE
process.title

### ISSUE THREAD

**Issue by oncletom** [oncletom/nodebook#9]
Spécifier un nom pour faciliter la lecture du processus dans `ps` et l'onglet du terminal.

cc @DavidBruant

**Comment by misterdjules** [comment]
FWIW, `process.name` does not exist, were you referring to `process.title`? If so:

1. `process.title` does not allow to change any terminal's window's title. To do that, one needs to use terminal escaping sequences.

2. Setting `process.title` is a confusing API to use. It does not behave the same on all platforms, and has some confusing limitations on all of them. I can expand on that if you'd like to get more details.

**Comment by oncletom** [comment]
@misterdjules correct, I was referring to `process.title` (bad memory).

Are there more limitations than the ones detailed on https://nodejs.org/api/process.html#process_process_title? More limitations than a size limit?

If you have time, I am curious to know more about it :-) (love reading insights from people working on the tools and languages themselves)

**Comment by misterdjules** [comment]
> Are there more limitations than the ones detailed on https://nodejs.org/api/process.html#process_process_title? More limitations than a size limit?

One additional limitation is that, currently, setting `process.title` is a no-op on some platforms like SmartOS and AIX, even though that is being fixed.

Another is that system tools on various platforms don't always read the arguments of a process from the memory area to which node writes when setting `process.title`. For instance, on SmartOS, running `ps -p pid -o args=` reads from `/proc/pid/psinfo` (which is not modified by setting `process.title`), but running `ps auxww` reads from the same memory area (although one _must_ pass at least two `ww` on the command line).

In other words, the resulting behavior is not consistent across platforms and even across system tools on a given platform.

**Comment by oncletom** [comment]
Thanks @misterdjules :-) your insights on less mainstream OS is definitely eye opening!

I think it would be good to mention this API and to state these limitations, mostly to tell people to use it only if necessary, while being extra careful.

**Comment by misterdjules** [comment]
> I think it would be good to mention this API and to state these limitations, mostly to tell people to use it only if necessary, while being extra careful.

Agreed.

**Comment by oncletom** [comment]
Thanks a lot @misterdjules for your very interesting knowledge :-)

Feel free to raise issues with unusual developer stuff or whatever you find interesting.
